src/Data/Monoid/Action.hs

```diff
@@ -81,3 +85,46 @@ instance Action m s => Action (Maybe m) s where
 instance Action (Endo a) a where
   act = appEndo
 
+instance Num a => Action Integer (Sum a) where
+  n `act` a = fromInteger n <> a
+
+instance Num a => Action Integer (Product a) where
+  n `act` a = fromInteger n <> a
+
+instance Fractional a => Action Rational (Sum a) where
+  n `act` a = Sum (fromRational n) <> a
+
+instance Fractional a => Action Rational (Product a) where
+  n `act` a = Product (fromRational n) <> a
+
+-- | An action of a group is "free transitive", "regular", or a "torsor"
+--   iff it is invertible.
+--
+--   Given a value `s2` that has been acted upon by `m`,
+--   and the original value `s1`, it is possible to recover the acting `m`.
+--   This is encoded in the law:
+--
+--   @
+--     (m `act` s) `difference` s = m
+--   @
+class Group m => Torsor m s where
+
+  -- | 'difference s1 s2' is the element @m@ such that @m `'act'` s1 = s2@.
```

Comment:
I think this comment is backwards.   The documentation of `Torsor` says ``(m `act` s) `difference` s = m`` (which makes sense to me so that `difference` acts just like the usual subtraction operator, *e.g.* `(2 + 5) - 5 = 2`).  But the documentation of `difference` implies that ``s1 `difference` (m `act` s1) = m``.
```suggestion
  -- | 'difference s2 s1' is the element @m@ such that @m `'act'` s1 = s2@.
```